std/json.d

```diff
@@ -359,6 +359,90 @@ struct JSONValue
         return type == JSONType.null_;
     }
 
+    /***
+     * Generic type value getter
+     * A convenience getter that returns `JSONValue` as the specified D type.
+     * Note: only arithmetic, `bool`, `string`, `JSONValue` and `JSONValue[]` types are accepted
+     * Throws: `JSONException` if `T` is not a valid type or conversion is impossible
+     */
+    @property inout(T) get(T)() inout const pure @safe
+    {
+        static if (is(T : string))
+        {
+            return str;
+        }
+        else static if (is(T : bool))
+        {
+            return boolean;
+        }
+        else static if (isFloatingPoint!T)
+        {
+            switch (type)
+            {
+            case JSONType.float_:
+                return cast(T)floating;
+            case JSONType.uinteger:
+                return cast(T)uinteger;
+            case JSONType.integer:
+                return cast(T)integer;
+            default:
+                throw new JSONException("JSONValue is not a number type");
+            }
+        }
+        else static if (__traits(isUnsigned, T))
+        {
+            return cast(T)uinteger;
+        }
+        else static if (__traits(isArithmetic, T))
+        {
+            return cast(T)integer;
+        }
+        else
+        {
+            throw new JSONException("Unsupported type");
+        }
+    }
+    // This specialization is needed because arrayNoRef requires inout
+    @property inout(T) get(T : JSONValue[])() inout pure @trusted
+    {
+        return arrayNoRef;
+    }
+    // ditto
+    @property inout(T) get(T : JSONValue[string])() inout pure @trusted
+    {
+        return object;
+    }
+    ///
+    @safe unittest
+    {
+        import std.exception;
+        string s =
+        `{
+            "a": 123,
+            "b": 3.1415,
+            "c": "text",
+            "d": true,
+            "e": [1, 2, 3],
+            "f": { "a": 1 }
+         }`;
+
+        struct a { }
+
+        immutable auto json = parseJSON(s);
```

Comment:
```suggestion
        immutable json = parseJSON(s);
```

In D you don't need to use `auto` for type inference - any storage class / type qualifier will work.

Reply:
Good idea, thank you!